### PR TITLE
apim86 - Add cache support

### PR DESCRIPTION
**Pull request by wtrocki**
# Motivation 

Allow to cache requests for certain amount of time. Users can specify how long they should be cached:

screen shot 2016-11-05 at 2 50 04 pm

Ticket: #86

### CODE REVIEW

File: lib/requester.js [hunk 1]
```diff
@@ -1,136 +1,169 @@
...
-  r = request[method]({
-    url : requestObject.url,
-    headers : headers,
-    body : requestObject.body,
-    json :json
-  }, function(error, response, body){
-    if (error){
-      error.code = 400;
-      return cb(error);
+  checkCache(requestObject, function (err, cachedReponse) {
```
Comment (wtrocki) [inline lib/requester.js:71]: We just wrapping existing code here in `checkCache` method. Rest of the code is unchanged.

File: lib/requester.js [hunk 2]
```diff
@@ -1,136 +1,169 @@
...
+    if(cachedReponse){
```
Comment (cianclarke) [inline lib/requester.js:72]: It might be nice here to be able to indicate the response came from the cache. Maybe a header `X-Is-Cached` or similar?

Comment (wtrocki) [reply lib/requester.js:72]: Good idea. Doing it!

**Comment by wtrocki** [comment]
Closing as outdated. Change is valid, but project is no longer actively maintained.

**Pull request closed without merging**